core/app/finders/spree/option_values/find_available.rb

```diff
@@ -4,7 +4,7 @@ class FindAvailable
       include ProductFilterable
 
       def initialize(scope: OptionValue.spree_base_scopes, products_scope: Product.spree_base_scopes)
-        @scope = scope
+        @scope = scope.order(position: :asc)
```

Comment:
@aleksandarpetrushev we need to sort this by option's type position first, and later by option value position